contracts/mainnet/connectors/fluid-vaults/main.sol

```diff
@@ -0,0 +1,299 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+/**
+ * @title Fluid.
+ * @dev Lending & Borrowing.
+ */
+
+import {Helpers} from "./helpers.sol";
+import {Events} from "./events.sol";
+import {IVaultT4} from "./interface.sol";
+
+abstract contract FluidVaultT4Connector is Helpers, Events {
+    /**
+     * @dev Deposit, borrow, payback and withdraw assets from the vault.
+     * @notice Single function which handles supply, withdraw, borrow & payback
+     * @param vaultAddress_ Vault address.
+     * @param nftId_ NFT ID for interaction. If 0 then create new NFT/position.
+     * @param newColToken0_ Token 0 new collateral. If positive then deposit, if negative then withdraw, if 0 then do nothing.
+     * @param newColToken1_ Token 1 new collateral. If positive then deposit, if negative then withdraw, if 0 then do nothing.
+     * For max deposit use type(uint25).max, for max withdraw use type(uint25).min.
+     * @param colSharesMinMax_ Min or max collateral shares to mint or burn (positive for deposit, negative for withdrawal)
+     * @param newDebtToken0_ Token 0 new debt. If positive then borrow, if negative then payback, if 0 then do nothing
+     * @param newDebtToken1_ Token 1 new debt. If positive then borrow, if negative then payback, if 0 then do nothing
+     * For max payback use type(uint25).min.
+     * @param debtSharesMinMax_ Min or max debt shares to burn or mint (positive for borrowing, negative for repayment)
+     * @param repayApproveAmtToken0_ In case of max amount for payback, this amount will be approved for token 0 spending.
+     * @param repayApproveAmtToken1_ In case of max amount for payback, this amount will be approved for token 1 spending.
+     * Should always be positive.
+     * @param getIds_ Array of 9 elements to retrieve IDs.
+     * @param setIds_ Array of 9 elements to set IDs:
+     * Nft Id
+     * Supply amount token 0
+     * Supply amount token 1
+     * Withdraw amount token 0
+     * Withdraw amount token 1
+     * Borrow amount token 0
+     * Borrow amount token 1
+     * Payback amount token 0
+     * Payback amount token 1
+     */
+
+     // WITHDRAW AND PAYBACK
+    function operateWithIds(
+        address vaultAddress_,
+        uint256 nftId_,
+        int256 newColToken0_,
+        int256 newColToken1_,
+        int256 colSharesMinMax_,
+        int256 newDebtToken0_,
+        int256 newDebtToken1_,
+        int256 debtSharesMinMax_,
+        uint256 repayApproveAmtToken0_,
+        uint256 repayApproveAmtToken1_,
```

Comment:
Not needed. If `newDebtToken0_` & `newDebtToken1_` is < 0 then that means payback and approve amount should also be this amount but in uint (not int)

Reply:
Resolved